great_tables/gt.py

```diff
@@ -291,14 +319,37 @@ def _build_data(self, context: str) -> Self:
 
         return built._replace(_body=final_body, _stub=final_stub)
 
-    def render(self, context: str) -> str:
-        html_table = self._build_data(context=context)._render_as_html()
+    def render(
+        self,
+        context: str,
+        make_page: bool = False,
```

Comment:
If we want to give people the option to set this, but not require it (e.g. render feels like it can run "automatically", similar to `_repr_html_()`), we could set arguments like `make_page=None` by default. In the case where `make_page` and `all_important` are both None, this might essentially produce the same behavior as `_repr_html_()`.

See comment in `_repr_html_()`. If we separate environment detection and options fetching (e.g. get `make_page` and `all_important` for environment X) then we also should be able to use that code here.

Reply:
From pairing, it sounds like we might want to use the following strategy:

* in the future, `.render()` might take a dataclass as context. The dataclass will hold rendering options (e.g.  LatexContext)
* in the now, `.as_raw_html()` is essentially the context specific version. The parameters in `.as_raw_html()` are what the options in the HtmlContext would be, etc..